File: sdks/java/lib/src/main/java/models/shared/RetentionPolicyAssignment.java
package .models.shared;

import java.time.OffsetDateTime;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import .utils.DateTimeSerializer;
import .utils.DateTimeDeserializer;

/**
 * RetentionPolicyAssignment
 * A retention assignment represents a rule specifying
 * the files a retention policy retains.
 * Assignments can retain files based on their folder or metadata,
 * or hold all files in the enterprise.
**/
public class RetentionPolicyAssignment {
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("assigned_at")
    public OffsetDateTime assignedAt;
    public RetentionPolicyAssignment withAssignedAt(OffsetDateTime assignedAt) {
        this.assignedAt = assignedAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("assigned_by")
    public UserMini assignedBy;
    public RetentionPolicyAssignment withAssignedBy(UserMini assignedBy) {
        this.assignedBy = assignedBy;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("assigned_to")
    public RetentionPolicyAssignmentAssignedTo assignedTo;
    public RetentionPolicyAssignment withAssignedTo(RetentionPolicyAssignmentAssignedTo assignedTo) {
        this.assignedTo = assignedTo;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("filter_fields")
    public RetentionPolicyAssignmentFilterFields[] filterFields;
    public RetentionPolicyAssignment withFilterFields(RetentionPolicyAssignmentFilterFields[] filterFields) {
        this.filterFields = filterFields;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("id")
    public String id;
    public RetentionPolicyAssignment withId(String id) {
        this.id = id;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("retention_policy")
    public RetentionPolicyMini retentionPolicy;
    public RetentionPolicyAssignment withRetentionPolicy(RetentionPolicyMini retentionPolicy) {
        this.retentionPolicy = retentionPolicy;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("start_date_field")
    public String startDateField;
    public RetentionPolicyAssignment withStartDateField(String startDateField) {
        this.startDateField = startDateField;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("type")
    public RetentionPolicyAssignmentTypeEnum type;
    public RetentionPolicyAssignment withType(RetentionPolicyAssignmentTypeEnum type) {
        this.type = type;
        return this;
    }
}

File: sdks/go/pkg/models/shared/retentionpolicyassignment.go
package shared

import (
	"time"
)

type RetentionPolicyAssignmentAssignedToTypeEnum string

const (
	RetentionPolicyAssignmentAssignedToTypeEnumFolder           RetentionPolicyAssignmentAssignedToTypeEnum = "folder"
	RetentionPolicyAssignmentAssignedToTypeEnumEnterprise       RetentionPolicyAssignmentAssignedToTypeEnum = "enterprise"
	RetentionPolicyAssignmentAssignedToTypeEnumMetadataTemplate RetentionPolicyAssignmentAssignedToTypeEnum = "metadata_template"
)

// RetentionPolicyAssignmentAssignedTo
// The `type` and `id` of the content that is under
// retention. The `type` can either be `folder`
// `enterprise`, or `metadata_template`.
type RetentionPolicyAssignmentAssignedTo struct {
	ID   *string                                      `json:"id,omitempty"`
	Type *RetentionPolicyAssignmentAssignedToTypeEnum `json:"type,omitempty"`
}

type RetentionPolicyAssignmentFilterFields struct {
	Field *string `json:"field,omitempty"`
	Value *string `json:"value,omitempty"`
}

type RetentionPolicyAssignmentTypeEnum string

const (
	RetentionPolicyAssignmentTypeEnumRetentionPolicyAssignment RetentionPolicyAssignmentTypeEnum = "retention_policy_assignment"
)

// RetentionPolicyAssignment
// A retention assignment represents a rule specifying
// the files a retention policy retains.
// Assignments can retain files based on their folder or metadata,
// or hold all files in the enterprise.
type RetentionPolicyAssignment struct {
	AssignedAt      *time.Time                              `json:"assigned_at,omitempty"`
	AssignedBy      *UserMini                               `json:"assigned_by,omitempty"`
	AssignedTo      *RetentionPolicyAssignmentAssignedTo    `json:"assigned_to,omitempty"`
	FilterFields    []RetentionPolicyAssignmentFilterFields `json:"filter_fields,omitempty"`
	ID              *string                                 `json:"id,omitempty"`
	RetentionPolicy *RetentionPolicyMini                    `json:"retention_policy,omitempty"`
	StartDateField  *string                                 `json:"start_date_field,omitempty"`
	Type            *RetentionPolicyAssignmentTypeEnum      `json:"type,omitempty"`
}
